src/Microsoft.ML.Mkl.Components/VectorWhitening.cs

```diff
@@ -617,6 +620,57 @@ protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, b
                     };
                 return del;
             }
+            public bool CanSaveOnnx(OnnxContext ctx) => true;
+
+            public void SaveAsOnnx(OnnxContext ctx)
+            {
+                Host.CheckValue(ctx, nameof(ctx));
+                int numColumns = _parent.ColumnPairs.Length; //why are the multiple columns
+                for (int iinfo = 0; iinfo < numColumns; ++iinfo)
+                {
+                    string inputColumnName = _parent.ColumnPairs[iinfo].inputColumnName;
+                    if (!ctx.ContainsColumn(inputColumnName))
+                        continue;
+
+                    string outputColumnName = _parent.ColumnPairs[iinfo].outputColumnName;
+                    string srcVariableName = ctx.GetVariableName(inputColumnName);
+                    string dstVariableName = ctx.AddIntermediateVariable(_srcTypes[iinfo], outputColumnName, true);
+                    SaveAsOnnxCore(ctx, iinfo, srcVariableName, dstVariableName);
+                }
+
+            }
+            private void SaveAsOnnxCore(OnnxContext ctx, int iinfo, string srcVariableName, string dstVariableName)
+            {
+                var model = _parent._models[iinfo];
+                int dimension = _srcTypes[iinfo].GetValueCount();
+                Host.Assert(model.Length == dimension * dimension);
+
+                var parameters = _parent._columns[iinfo];
+                Host.Assert(parameters.Kind == WhiteningKind.PrincipalComponentAnalysis || parameters.Kind == WhiteningKind.ZeroPhaseComponentAnalysis);
+
+                int rank = (parameters.Kind == WhiteningKind.PrincipalComponentAnalysis && parameters.Rank > 0) ? parameters.Rank : dimension;
+                long[] modelDimension = { rank, dimension };
+
+                if (rank != dimension)
+                {
+                    float[] principalComponents = new float[rank * dimension];
+                    for (int i = 0; i < parameters.Rank; i++)
+                    {
+                        if (i >= dimension)
+                            break;
+                        Array.Copy(model, i*dimension, principalComponents, i*dimension, dimension);
+                    }
+                    model = principalComponents;
+                }
+
+                var opType = "Gemm";
```

Comment:
>Gemm [](start = 30, length = 4)

Just out of curiosity: The `Mapper` of this class calls `Mkl.Gemv`, as opposed to `Gemm`. Is the ONNX operation `Gemm` equivalent to the MKL operation `Gemv`?

Reply:
Yes, they perform the same operation, with slightly different options.